bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/DirectoryDialog.java

```diff
@@ -123,9 +125,36 @@ public String getMessage () {
  * </ul>
  */
 public String open() {
+	try {
+		return askSelectDirectory().orElse(null);
+	} catch (SWTException e) {
+		if (e.code == SWT.ERROR_INVALID_RETURN_VALUE) {
+			return null;
+		}
+		throw e;
+	}
+}
+
+/**
+ * Makes the dialog visible and brings it to the front of the display.
+ * Equal to {@link DirectoryDialog#open()} but also exposes for state information like user cancellation.
+ *
+ * @return an Optional that either contains the absolute path of the selected directory
+ *         or is empty in case the dialog was canceled
+ *
+ * @exception SWTException <ul>
+ *    <li>ERROR_WIDGET_DISPOSED - if the dialog has been disposed</li>
+ *    <li>ERROR_THREAD_INVALID_ACCESS - if not called from the thread that created the dialog</li>
+ *    <li>ERROR_INVALID_RETURN_VALUE - if the dialog was not cancelled and did not return a valid path</li>
+ * </ul>
+ *
+ * @since 3.125
+ */
+public Optional<String> askSelectDirectory() {
```

Comment:
```suggestion
public Optional<String> openDirectory() {
```

What do you think about something like this? askSelect sounds a bit strange and that way probabbly if it also starts with "open" users that are used to the old method might discover the new one.

Reply:
Yes, I think something that starts with open so that it's sorted to be adjacent in documentation or in proposals....

Reply:
Agreed. Maybe just `openDialog()`? Since the call does not open a file/directory, but a dialog (to choose one).

Reply:
I would be fine with openDialog() as well